### DISCUSSION TITLE
Note not found page has status code 200

### DISCUSSION THREAD

**Discussion by MaienM** [srid/emanote#432] [Q&A]
**Describe the bug**
Note not found page has status code `200`.

**To Reproduce**
1. Start live server.
2. Go to the url of a note that does not exist.
3. Get an error page.
4. Notice that the status code is `200`.

**Expected behavior**
The error pages should have appropriate status codes. For the node not found page this should probably be `404`, for the ambiguous note probably something in the 500 range.

**Screenshots**
![2023-04-03T10:53:52+0200-654x289](https://user-images.githubusercontent.com/583129/229472806-67f21800-61a7-48ca-b613-60614e7b5b1b.png)
![2023-04-03T11:03:31+0200-247x61](https://user-images.githubusercontent.com/583129/229472801-e60f53f6-38cf-4900-88f4-29f12ad98beb.png)

**Desktop (please complete the following information):**
 - Firefox 111.0.1
 - Emanote 1.0.2.0

**Additional context**
N/A

**Comment by srid** [comment]
Are you using the live server for serving production site (which it is not meant for)? What should happen during [hot reload](https://ema.srid.ca/topics/hot-reload) -- as opposed to manual page load? Anyway, this is all more a discussion for https://github.com/srid/ema

**Reply by MaienM** [reply]
No, not using it for production, just locally. I imagine this issue probably doesn't exists in production; I haven't actually setup a production version yet.

I ran into this because I'm adding some javascript to the site, and got unexpected results while loading some resources that only exist for some of the pages. I've worked around it by inspecting the result itself (which should be an SVG document in this case, so it's easy to distinguish from an HTML response), but I was surprised to see that the response for "this doesn't exist" is `200`; I had blindly assumed it would be `404`.

WRT hot reloading, that's a good question. DOM patching wouldn't really work in this case, so perhaps switching to/from an error page should trigger a page reload instead?

I don't mind moving this to the ema repo if you'd prefer that, LMK if you'd like me to re-post my messages there.

**Reply by srid** [reply]
> I don't mind moving this to the ema repo if you'd prefer that, LMK if you'd like me to re-post my messages there.

Sounds good to me.
